gradle-recommended-product-dependencies/src/main/java/com/palantir/gradle/dist/CompileRecommendedProductDependencies.java

```diff
@@ -33,20 +35,25 @@ public abstract class CompileRecommendedProductDependencies extends DefaultTask
     @Input
     abstract SetProperty<ProductDependency> getRecommendedProductDependencies();
 
+    /**
+     * Ensure that the sourcesJar task in {@link RecommendedProductDependenciesPlugin} includes {@code getProductDependenciesFile()}.
+     */
     @OutputDirectory
     abstract DirectoryProperty getOutputDir();
 
-    @TaskAction
-    final void action() throws IOException {
-        File outputFile = getOutputDir()
-                .get()
+    @Internal
+    public final File getProductDependenciesFile() {
```

Comment:
This should probably be a `Provider`, no?

Also, any reason not to make this an `@OutputFile`? That way Gradle will automatically create the parent directories.